lms/djangoapps/instructor_task/tasks_helper.py

```diff
@@ -1015,6 +1015,45 @@ def _order_problems(blocks):
     return problems
 
 
+def parse_student_data(student_data):
+    """
+    Parses student data to output a more easily-readable version of the 'state', which includes last submission time,
+    number of attempts and the students' answers for a specific problem
+
+    :param student_data: dict with 2 keys: 'username' and state'. The value of 'state' is a string version of a dict
+
+    :return: 2-d array, where a row is the username and state data for a student. First row is the headers. Only
+        includes students that have submitted an answer to the problem
+    """
+    headers = ['username']
+    rows = []
+
+    # populate header fields if there is at least one student that has submitted an answer
+    for student in student_data:
+        state = student['state']
+        d = json.loads(state)  # state is a string
+        if 'student_answers' in d:  # we're looking at a student that has submitted answers
```

Comment:
This won't work for all xblocks. The naming of the field is author specified and arbitrary. For example capa probs write "student_answers" but the submit-and-compare xblock writes "student_answer". Sad but true.